web3/hardhat/contracts/Crowdfund.sol

```diff
@@ -0,0 +1,134 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+error NotEnoughETHDonation();
+error NotEnoughBalance();
+error FailedToSendNativeToken();
+
+contract Crowdfund is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    IERC20 public token;
+    uint256 public startsAt;
+    uint256 public endsAt;
+    uint256 public goal;
+    bytes public metaPtr;
+
+    bool public canRefund;
+
+    mapping(address => uint256) userFunds;
+
+    event donated(address sender, uint256 amount);
+    event funded(address to, uint256 amount);
+    event refundActive(bool refunding);
+    event userBalanceWithdrawn(address user, uint256 balance);
```

Comment:
Use captial letters for events:
https://docs.soliditylang.org/en/v0.8.19/style-guide.html#event-names

Reply:
Thanks for this, done